ArmaforcesMissionBot/Handlers/ReactionHandler.cs

```diff
@@ -0,0 +1,320 @@
+using ArmaforcesMissionBot.DataClasses;
+using Discord;
+using Discord.WebSocket;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading.Tasks;
+using System.Timers;
+using ArmaforcesMissionBot.Helpers;
+using Microsoft.CodeAnalysis.CSharp.Syntax;
+using static ArmaforcesMissionBotSharedClasses.Mission;
+using Microsoft.CodeAnalysis.Differencing;
+using CSharpFunctionalExtensions;
+
+namespace ArmaforcesMissionBot.Handlers
+{
+    public static class StringExtensionMethods
+    {
+        public static string ReplaceFirst(this string text, string search, string replace)
+        {
+            int pos = text.IndexOf(search);
+            if (pos < 0)
+            {
+                return text;
+            }
+            return text.Substring(0, pos) + replace + text.Substring(pos + search.Length);
+        }
+    }
+    public class ReactionHandler : IInstallable
+    {
+        private DiscordSocketClient _client;
+        private MiscHelper _miscHelper;
+        private IServiceProvider _services;
+        private Config _config;
+        private Timer _timer;
+
+        public async Task Install(IServiceProvider map)
+        {
+            _client = map.GetService<DiscordSocketClient>();
+            _config = map.GetService<Config>();
+            _miscHelper = map.GetService<MiscHelper>();
+            _services = map;
+            // Hook the MessageReceived event into our command handler
+            _client.ReactionAdded += HandleReactionAdded;
+            _client.ReactionRemoved += HandleReactionRemoved;
+
+            _timer = new Timer();
+            _timer.Interval = 2000;
+
+            _timer.Elapsed += CheckReactionTimes;
+            _timer.AutoReset = true;
+            _timer.Enabled = true;
+        }
+
+        private async Task HandleReactionAdded(Cacheable<IUserMessage, ulong> message, ISocketMessageChannel channel, SocketReaction reaction)
```

Comment:
Please "split" this method in half. One for role reactions and the other for signups reactions. And here just determine which one should be used.

The same goes for reaction removed.

Reply:
Done.
One note - animated emojis probably don't work. I can't test it.

Reply:
I believe they never did as bot can't add them.